src/fsharp/FSharp.Core/seqcomposer.fs

```diff
@@ -1006,12 +1002,12 @@ namespace Microsoft.FSharp.Collections
                         Unchecked.defaultof<_> (* return value unused in Fold context *) })
 
         [<CompiledName "Except">]
-        let inline except (itemsToExclude: seq<'T>) (source:ISeq<'T>) : ISeq<'T> when 'T:equality =
+        let inline except (itemsToExclude:seq<'T>) (source:ISeq<'T>) : ISeq<'T> when 'T:equality =
             source.PushTransform { new TransformFactory<'T,'T>() with
                 override __.Compose _ _ next =
-                    upcast { new Transform<'T,'V,Lazy<HashSet<'T>>>(next,lazy(HashSet<'T>(itemsToExclude,HashIdentity.Structural<'T>))) with
+                    upcast { new Transform<'T,'V,HashSet<'T>>(next,HashSet<'T>(itemsToExclude,HashIdentity.Structural<'T>)) with
```

Comment:
I'm don't think removing `Lazy` here is the correct thing to do. The original code is:

        [<CompiledName("Except")>]
        let except (itemsToExclude: seq<'T>) (source: seq<'T>) =
            checkNonNull "itemsToExclude" itemsToExclude
            checkNonNull "source" source

            seq {
                use e = source.GetEnumerator()
                if e.MoveNext() then
                    let cached = HashSet(itemsToExclude, HashIdentity.Structural)
                    let next = e.Current
                    if (cached.Add next) then yield next
                    while e.MoveNext() do
                        let next = e.Current
if (cached.Add next) then yield next }

Where it doesn't create the `HashSet` until after the first call to `MoveNext`. So if the code never enumerates then the it is never allocated. But this was done to fix a unit test? Which test was it?